docs/docs/guides/04-fe-coding-guidelines.mdx

```diff
@@ -0,0 +1,596 @@
+render as you fetch. For mutations we'll be using `@redux/toolkit` and subsequently it's caching
+mechanisms making `react-query` redundant.
+
+:::note
+🚧 This section will be filled in further upon delivery of v5.
+:::
+
+## Managing state
+
+### Local / Global / URL state
+
+In the following paragraph we discuss three different types of state – Local which commonly is
+defined by using `React.useState`, Global which in our case involves using `redux` and URL based
+state e.g. a path parameter or search params.
+
+**Local State**
+
+This is the typical place to start when you need to make your app “react” to a change, for example,
+through event inputs e.g “A user presses a button and a modal opens”:
+
+```jsx
+import { useState } from 'react';
+
+const MyButton = () => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleClick = () => setIsOpen((currOpen) => !currOpen);
+
+  return (
+    <button type="button" onClick={handleClick}>
+      {isOpen ? 'Close modal' : 'Open modal'}
+    </button>
+  );
+};
+```
+
+Above is a simple example of using local state, it's isolated to the component and we're not
+typically passing this further down the tree. It's important to remember that you can very easily
+derive state from either your current state or props that are passed e.g. you render a table of
+entries and you want to know the count of entries that are “published” you could create a new state
+value, but that's not necessary, we can do a calculation because the root of the data is based off
+the props thus when the props update so will the count:
+
+```jsx
+const MyTable = ({ entries = [] }) => {
+  const publishedCount = entries.filter((e) => e.isPublished).length;
+
+  return (
+    <div>
+      <p>You have {publishedCount} published entries</p>
+      <List entries={entries} />
+    </div>
+  );
+};
+```
+
+Always aim to derive state as opposed to setting state where possible!
+
+**Global state (redux)**
+
+When's the right time to use global state? Well first if we look at how you can pass state around
+your app we first look at prop drilling. This can become verbose and inconvenient when you start to
+pass it down past the first child. Often composability can be incredibly useful for these scenarios
+where this:
+
+```jsx
+import { useState } from 'react';
+
+const App = () => {
+  const [someState, setSomeState] = useState('some state');
+  return (
+    <>
+      <Header someState={someState} onStateChange={setSomeState} />
+      <LeftNav someState={someState} onStateChange={setSomeState} />
+      <MainContent someState={someState} onStateChange={setSomeState} />
+    </>
+  );
+};
+```
+
+can easily become this:
+
+```jsx
+import { useState } from 'react';
+
+const App = () => {
+  const [someState, setSomeState] = useState('some state');
+  return (
+    <>
+      <Header
+        logo={<Logo someState={someState} />}
+        settings={<Settings onStateChange={setSomeState} />}
+      />
```

Comment:
Don't we want to encourage this pattern? Or maybe you are trying to illustrate multiple options?

```suggestion
      <Header>
      	<Logo someState={someState} />
      	<Settings onStateChange={setSomeState} />
      </Header>
```

Reply:
I think you're right, lets not advertise this it's confusing ✅